examples/features/retry/client/main.go

```diff
@@ -47,16 +47,19 @@ var (
 		}]}`
 )
 
-// use grpc.WithDefaultServiceConfig() to set service config
-func retryDial() (*grpc.ClientConn, error) {
+// newClientWithRetries uses grpc.WithDefaultServiceConfig() to set
+// service config, and creates the channel. However, the recommended
+// approach is to fetch the retry configuration from the name resolver
+// rather than defining it on the client side.
+func newClientWithRetries() (*grpc.ClientConn, error) {
```

Comment:
I don't understand the need for this function `newClientWithRetries`. I would recommend getting rid of it, and directly calling `grpc.NewClient` from `main`.